src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java

```diff
@@ -0,0 +1,195 @@
+     * @param hand the hand
+     * @return the result
+     */
+    @Override
+    @NotNull
+    public ActionResult<ItemStack> onItemRightClick(
+            @NotNull ItemStack scepter,
+            World worldIn,
+            EntityPlayer playerIn,
+            EnumHand hand)
+    {
+        if (worldIn.isRemote)
+        {
+            return new ActionResult(EnumActionResult.PASS, scepter);
+        }
+
+        if (!scepter.hasTagCompound())
+        {
+            scepter.setTagCompound(new NBTTagCompound());
+        }
+        final NBTTagCompound compound = scepter.getTagCompound();
+
+        toggleItemMode(playerIn, compound);
+
+        return new ActionResult(EnumActionResult.SUCCESS, scepter);
+    }
+
+    @NotNull
+    private EnumActionResult handleItemAction(
+            NBTTagCompound compound,
+            EntityPlayer playerIn,
+            World worldIn,
+            BlockPos pos,
+            ColonyView colonyView)
+    {
+        String tagItemMode = compound.getString(TAG_ITEM_MODE);
+
+        switch(tagItemMode) {
+            case TAG_VALUE_MODE_BLOCK:
+                return handleAddBlockType(playerIn, worldIn, pos, colonyView);
+            case TAG_VALUE_MODE_LOCATION:
+                return handleAddLocation(playerIn, worldIn, pos, colonyView);
+            default:
+                toggleItemMode(playerIn, compound);
+                return handleItemAction(compound, playerIn, worldIn, pos, colonyView);
+        }
+    }
+
+    @NotNull
+    static private void toggleItemMode(EntityPlayer playerIn, NBTTagCompound compound)
+    {
+        String itemMode = compound.getString(TAG_ITEM_MODE);
+
+        switch (itemMode)
+        {
+            case TAG_VALUE_MODE_BLOCK:
+                compound.setString(TAG_ITEM_MODE, TAG_VALUE_MODE_LOCATION);
+                LanguageHandler.sendPlayerMessage(playerIn, "com.minecolonies.coremod.item.permissionscepter.setmode", "location");
+            case TAG_VALUE_MODE_LOCATION:
+            default:
+                compound.setString(TAG_ITEM_MODE, TAG_VALUE_MODE_BLOCK);
+                LanguageHandler.sendPlayerMessage(playerIn, "com.minecolonies.coremod.item.permissionscepter.setmode", "block");
+        }
+    }
+
+    @NotNull
+    static private EnumActionResult handleAddBlockType(
+            final EntityPlayer playerIn,
+            final World worldIn,
+            final BlockPos pos,
+            final ColonyView colonyView)
+    {
+        IBlockState blockState = worldIn.getBlockState(pos);
+        Block block = blockState.getBlock();
+
+        ChangeFreeToInteractBlockMessage message = new ChangeFreeToInteractBlockMessage(
+                colonyView,
+                block,
+                ChangeFreeToInteractBlockMessage.MessageType.ADD_BLOCK);
+        MineColonies.getNetwork().sendToServer(message);
+
+        return EnumActionResult.SUCCESS;
+    }
+
+    @NotNull
+    static private EnumActionResult handleAddLocation(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)